### PR TITLE
added support for generating go code (only locally so far)

### PR DESCRIPTION
**Pull request by moritzzimmer**
- enhanced Makefile
- documented tools for workflow on MacOS
- added generated files since they need to live somewhere for go's package management

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -29,6 +29,12 @@ ifeq ($(LANGUAGE),node)
 	FLAGS+= --js_out=import_style=commonjs,binary:$(OUTPUT)
 	FLAGS+= --ts_out=service=grpc-node:$(OUTPUT)
 	FLAGS+= --grpc_out=$(OUTPUT)
+else ifeq ($(LANGUAGE),go)
+	ifeq ($(OUTPUT), $(JAVA_DIR))
+		OUTPUT = $(GO_DIR)
+	endif
+	FLAGS+= --$(LANGUAGE)_out=plugins=grpc:$(OUTPUT)
```
Comment (harryurban) [inline Makefile:36]: Linter würde sagen “value of LANGUAGE is always ‘go’”

File: stroeer/web/article/v1/web_article.proto [hunk 2]
```diff
@@ -4,6 +4,7 @@ package stroeer.web.article.v1;
 
 import "stroeer/core/v1/article.proto";
 
+option go_package = "github.com/stroeer/tapir/web/article/v1;article";
```
Comment (shobhitsharma) [inline stroeer/web/article/v1/web_article.proto:7]:
```suggestion
option go_package = "github.com/stroeer/tapir/go/web/article/v1;article";
```

File: stroeer/web/article/v1/web_article_service.proto [hunk 3]
```diff
@@ -4,6 +4,7 @@ package stroeer.web.article.v1;
 
 import "stroeer/web/article/v1/web_article.proto";
 
+option go_package = "github.com/stroeer/tapir/web/article/v1;article";
```
Comment (shobhitsharma) [inline stroeer/web/article/v1/web_article_service.proto:7]:
```suggestion
option go_package = "github.com/stroeer/tapir/go/web/article/v1;article";
```

File: stroeer/web/section/v1/web_section_service.proto [hunk 4]
```diff
@@ -4,6 +4,7 @@ package stroeer.web.section.v1;
 
 import "stroeer/web/article/v1/web_article.proto";
 
+option go_package = "github.com/stroeer/tapir/web/section/v1;section";
```
Comment (shobhitsharma) [inline stroeer/web/section/v1/web_section_service.proto:7]:
```suggestion
option go_package = "github.com/stroeer/tapir/go/web/section/v1;section";
```

File: stroeer/web/article/v1/web_article.proto (hunk 2 above)
Comment (shobhitsharma) [inline stroeer/web/article/v1/web_article.proto:7]: @moritzzimmer Shouldn't this resolve the generate go files from `protoc`'s `--go_out` directory?

```suggestion
option go_package = "github.com/stroeer/tapir/go/stroeer/web/article/v1;article";
```

File: go/go.mod [hunk 5]
```diff
@@ -0,0 +1,9 @@
+module github.com/stroeer/tapir/go
```
Comment (shobhitsharma) [inline go/go.mod:1]:
```bash
$ go mod tidy
```

Outputs:

```bash
go: finding module for package github.com/stroeer/tapir/web/article/v1
go: finding module for package github.com/stroeer/tapir/core/v1
github.com/stroeer/tapir/go/stroeer/web/article/v1 imports
        github.com/stroeer/tapir/core/v1: module github.com/stroeer/tapir@latest found (v0.0.0-20200528112422-d49d1b0955fa), but does not contain package github.com/stroeer/tapir/core/v1
github.com/stroeer/tapir/go/stroeer/web/section/v1 imports
        github.com/stroeer/tapir/web/article/v1: module github.com/stroeer/tapir@latest found (v0.0.0-20200528112422-d49d1b0955fa), but does not contain package github.com/stroeer/tapir/web/article/v1
```

Comment (shobhitsharma) [reply go/go.mod:1]: Probably resolve it while generating go code with `option go_package`?

File: Makefile (hunk 1 above)
Comment (shobhitsharma) [reply Makefile:36]: Isn't that in LANGUAGE=go condition? Should be fine.

**Pull request merged**